ksqldb-common/src/main/java/io/confluent/ksql/logging/processing/ProcessingLoggerImplWithMetrics.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logging.processing;
+
+import java.util.Map;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.CumulativeSum;
+
+public class ProcessingLoggerImplWithMetrics extends ProcessingLoggerImpl {
```

Comment:
Let's model this as a composition relationship rather than inheritance:
```
interface MeteredProcessinLogger implements ProcessingLogger {
    ProcessingLogger inner;

    @Override
    public void error(final ErrorMessage msg) {
        totalProcessingErrors.record(0.5)
        inner.error(msg);
    }
}
```

Reply:
done